internal/ctrlutils/ingress-status.go

```diff
@@ -142,13 +154,18 @@ func (s *StatusUpdater) PullConfigUpdate(
 						s.log.Error(err, "failed to look up status info for Kong proxy Service", "service", s.publishService)
 					}
 				}
+				go func() {
+					time.Sleep(time.Minute * 1)
+					s.kongConfig.ConfigDone <- updateDone
+					s.log.V(util.DebugLevel).Info("retrying update", "timestamp", updateDone.Timestamp)
+				}()
```

Comment:
It appears this function is at the center of the fix we're trying to provide here as it makes sure that a config update doesn't go out of band. While we expect eventual consistency herein it could theoretically run indefinitely (or at least very long in the case of a slow GKE LB) which I understand to be the reason we've added a `time.Sleep` but I think the minute timing is a bit too long for two reasons:

1. at least currently the updates that result here are effectively for presentation/cosmetics, so we expect a human is watching
2. re-queuing the update is very low cost since we will simply throw it away if it becomes outdated

As such, I'm OK with a slight sleep here so that we're reducing log spam, but I think something with a shorter turnaround might be more ideal:

```suggestion
				go func() {
					time.Sleep(time.Second * 3)
					s.kongConfig.ConfigDone <- updateDone
					s.log.V(util.DebugLevel).Info("retrying update", "timestamp", updateDone.Timestamp)
				}()
```

And then I think we should capture this value in a variable and document it:

```go
// configUpdateRetryWait is a delay that will occur before trying to re-publish
// a configuration update that could not be published because the statusConfig
// was not yet ready.
const configUpdateRetryWait = time.Second * 3
```

Reply:
Time left as-is. The longer wait is intended more for the case where the LB never actually provisions, in which some retry will continue logging indefinitely, and the log is useless spam.

For the case where a retry actually matters (LB slow to provision), it's a one-off event at startup on a fresh install. I raised the log level and added the duration to make it clearer what's happening.